presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -138,8 +142,8 @@ public Iterable<HivePartition> getPartitionsIterator(
         }
         else {
             return () -> {
-                List<String> filteredPartitionNames = getFilteredPartitionNames(session, metastore, tableName, effectivePredicate);
-                return filteredPartitionNames.stream()
+                List<String> partitionNames = partitionFilteringFromMetastoreEnabled ? getFilteredPartitionNames(session, metastore, tableName, effectivePredicate) : getAllPartitionNames(session, metastore, tableName, constraint);
```

Comment:
is it my understanding correct that you will use this config to disable the get filtered partition names? and getAllPartitionName will perform faster?

Reply:
getAllPartitionNames will not get faster - but the retrieved partition names can now be cached and used by subsequent queries to avoid a remote call to the metastore

Reply:
There are 3 apis to retrieve partitions from Glue (GetPartition, BatchGetPartition & GetPartitions). The final one is what is used in Presto today and it accepts an expression that may be used to filter out partitions that are not interesting. We do not know how it is implemented under the covers. 

I take your point that we may be better off using BatchGetPartitions when retrieving all the table partitions and this is worth further investigation/benchmarking. However I propose we do that as a follow on since that is tangential to this PR. This focus of this PR is to cache and reuse the previously fetched partitions to avoid multiple expensive apis to the metastore

Reply:
thanks, point taken. sounds reasonable with a config to toggle this as a safety.